gfx-glyph/src/shader/vert.glsl

```diff
@@ -11,6 +11,13 @@ in vec4 color;
 out vec2 f_tex_pos;
 out vec4 f_color;
 
+const mat4 INVERT_Y_AXIS = mat4(
+    vec4(1.0, 0.0, 0.0, 0.0),
+    vec4(0.0, -1.0, 0.0, 0.0),
+    vec4(0.0, 0.0, 1.0, 0.0),
+    vec4(0.0, 0.0, 0.0, 1.0)
+);
```

Comment:
I have moved the y axis transformation to the shader, I think it makes things clearer when calculating tex coordinates.

Reply:
This seems to make the shader more complicated though, why not do this in the projection? `ortho(0.0, width, height, 0.0, 1.0, -1.0)`

Reply:
That works too, but we will have to remind the users of `gfx_glyph` about it. I think this is an implementation detail and should not leak.

Reply:
I guess I'm just not sure why one is more "right" than the other. The top as 0, and the bottom as height seems to match the default layout style.

But are you saying this is backwards from what you'd expect elsewhere?

Reply:
To be clear; I'm fairly happy to go with either way really, coord conversion gives me a headache so you probably have a better perspective on it.

Reply:
Yeah, I am currently struggling to figure it out :sweat_smile: Vulkan uses different normalized coordinates. Let me report back in a bit.

Reply:
Ah I think I understand, the `to_vertex` function now inverts the y dimension I think. The rusttype texture cache gives pixel coords
> These pixel-space coordinates assume an origin at the top left of the quad.

I think we can switch the vertex coords round and have the ortho projection as before

Reply:
I think I got it.

An orthographic projection assumes the Y axis for the clipping planes coordinates grows upwards, independently of the actual rendering coordinates. That's why when `bottom` is less than `top`, we get an inverted Y axis:

![image](https://user-images.githubusercontent.com/518289/56978128-18087b00-6b77-11e9-983f-f50576e5d84a.png)

See the `(2 / (top - bottom))`. Providing `top` as `height` and `bottom` as `0.0  basically leaks the fact that the OpenGL Y axis is backwards, we should make that transformation internally I think.

Reply:
In other words, the current orthographic projection should invert the axis, not produce the correct result.

Reply:
Yep although users of gfx_glyph _are_ opengl users, but I'm cool with either.

It actually might not matter if we provide the default transform with:
```rust
pub fn default_transform(target: &CV) -> [[f32; 4]; 4] {...}
```

This way it's internal either way

Reply:
> Yep although users of gfx_glyph are opengl users

Yes, but text positions are all provided assuming an inverted Y axis. I don't think it makes sense to ask users to invert the axis themselves. A transform should be consistent with the exposed coordinate space.

> It actually might not matter if we provide the default transform

Personally, I won't be using the default transform. My engine can render multiple things besides text, so I already have an orthographic projection ready.

I think the idea would be to keep transforms consistent between different `glyph_brush` backends, just the same way text positions should be consistent. This way, I can easily use the same transform in `wgpu_glyph` and `gfx_glyph` in my engine.

I do not mind much either way, as I can just invert the axis manually when using `gfx_glyph`, but I personally would expect the transform to be consistent with the coordinate space that the whole API works in.

Reply:
Ok I'll go with you